scripts/migrate-users.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import argparse
+from datetime import datetime
+import sys
+import json
+import yaml
+import urllib
+import requests
+import time
+
+# import logging
+# import httplib as http_client
+# http_client.HTTPConnection.debuglevel = 1
+# logging.basicConfig()
+# logging.getLogger().setLevel(logging.DEBUG)
+# requests_log = logging.getLogger("requests.packages.urllib3")
+# requests_log.setLevel(logging.DEBUG)
+# requests_log.propagate = True
+
+# If you're running into M_NOT_JSON issues, python-requests strips the body of request on 301 redirects. Make sure you're using the direct url of your homeserver.
+# See https://github.com/requests/requests/issues/2590
+
+def get_appservice_token(reg):
+    with open(reg, "r") as f:
+        reg_yaml = yaml.load(f)
+        return reg_yaml["as_token"]
+
+def get_users(homeserver, room_id, token, user_prefix, name_suffix):
+    res = requests.get(homeserver + "/_matrix/client/r0/rooms/" + urllib.quote(room_id) + "/joined_members?access_token=" + token)
+    joined = res.json().get("joined", None)
+    user_ids = [user_id for user_id in joined if user_id.startswith(user_prefix) and (joined.get(user_id).get("display_name") or "").endswith(name_suffix) ]
+    return { uid: joined.get(uid).get("display_name") for uid in user_ids }
+
+def get_rooms(homeserver, token):
+    res = requests.get(homeserver + "/_matrix/client/r0/sync?access_token=" + token).json()
+    room_ids = []
+    for room_id in res["rooms"]["join"]:
+        room = res["rooms"]["join"][room_id]
+        room_ids.append(room_id)
+    return room_ids
+
+def migrate_displayname(uid, oldname, suffix, homeserver, token):
+    newname = oldname.replace(suffix, "")
+    print("Migrating %s from %s to %s" % (uid, oldname, newname))
+    headers = { 'Content-Type': 'application/json' }
+    res = requests.put(homeserver + "/_matrix/client/r0/profile/" + urllib.quote(uid) + "/displayname?access_token=" + token + "&user_id=" + urllib.quote(uid),
+                       data = json.dumps({ 'displayname': newname }), headers=headers)
+    if res.json():
+        print(res.json())
+        if 'M_NOT_JSON' in str(res.json()):
+            print("python-requests strips the body of the request on 301 redirects (https://github.com/requests/requests/issues/2590). Make sure you're using the direct url of your homeserver.")
+    time.sleep(0.1)
```

Comment:
Nit: I would prefer to have the sleep be a part of the for loop which calls `migrate_displayname` than have it here.